runtime/shared_common/shrinit.cpp

```diff
@@ -2720,14 +2722,20 @@ ensureCorrectCacheSizes(J9JavaVM *vm, J9PortLibrary* portlib, U_64 runtimeFlags,
 		*cacheSize = J9_SHARED_CLASS_CACHE_MAX_SIZE;
 	}
 
+	U_64 maxSize = 0;
 	if (J9PORT_SHR_CACHE_TYPE_NONPERSISTENT == getCacheTypeFromRuntimeFlags(runtimeFlags)) {
-		U_64 maxSize = 0;
-
 		if ((J9PORT_LIMIT_LIMITED == j9sysinfo_get_limit(J9PORT_RESOURCE_SHARED_MEMORY, &maxSize))
 			&& (*cacheSize > maxSize)
 		) {
 			adjustCacheSizes(portlib, verboseFlags, piconfig, maxSize);
 		}
+	} else {
+		if (is64BitPlatDefaultSize) {
+			if (isFreeDiskSpaceLow(vm, &maxSize)) {
+				Trc_SHR_Assert_True(*cacheSize > maxSize);
+				adjustCacheSizes(portlib, verboseFlags, piconfig, maxSize);
+			}
```

Comment:
There should be tracepoints if isFreeDiskSpaceLow fails. Should we fallback to 64M if we can't check the free space?

Reply:
I will add the tracepoints inside isFreeDiskSpaceLow(). It is safer to use 64MB if we cannot check the free space.